pkg/streamingpromql/testdata/ours/limit.test

```diff
@@ -0,0 +1,331 @@
+  {} 8+0x10
+
+eval range from 0 to 50m step 5m count(limit_ratio(time() % 17/17, http_requests) and limit_ratio( - (1.0 - (time() % 17/17)), http_requests))
+# empty
+
+# Poor man's normality check: ok (loaded samples follow a nice linearity over labels and time).
+# The check giving: 1 (i.e. true).
+eval range from 0 to 50m step 5m abs(avg(limit_ratio(0.5, http_requests{instance!~"histogram_[0-9]"})) - avg(limit_ratio(-0.5, http_requests{instance!~"histogram_[0-9]"}))) <= bool stddev(http_requests{instance!~"histogram_[0-9]"})
+  {} 1+0x10
+
+# All specified histograms are included for r=1.
+eval instant at 50m limit_ratio(1, http_requests{instance="histogram_1"})
+  {__name__="http_requests", group="canary", instance="histogram_1", job="api-server"} {{count:10 sum:10}}
+
+eval range from 0 to 50m step 5m limit_ratio(1, http_requests{instance="histogram_1"})
+  {__name__="http_requests", group="canary", instance="histogram_1", job="api-server"} {{count:10 sum:10}}x10
+
+# Bug #15971.
+eval range from 0m to 50m step 5m count(limit_ratio(scalar(bar), http_requests))
+  {} _ 8 _ 8 _ 8 _ 8 _ 8 _
+
+clear
+
+load 1m
+  mixed{instance="1"} {{schema:0 sum:20 count:20}} {{schema:0 sum:20 count:20}} {{schema:0 sum:20 count:20}} _ 10 10 10
+  mixed{instance="2"} 10 10 10 _ {{schema:0 sum:20 count:20}} {{schema:0 sum:20 count:20}} {{schema:0 sum:20 count:20}}
+  foo 1 2 1 2 1 2 1 2 1 2 1 2
+
+eval instant at 0m limitk(1, mixed{instance="1"})
+  {__name__="mixed", instance="1"} {{schema:0 sum:20 count:20}}
+
+eval instant at 0m count(limitk(2, mixed))
+  {} 2
+
+eval instant at 2m limitk(1, mixed{instance="1"})
+  {__name__="mixed", instance="1"} {{schema:0 sum:20 count:20}}
+
+eval instant at 3m limitk(1, mixed{instance="1"})
+  {__name__="mixed", instance="1"} {{schema:0 sum:20 count:20}}
+
+eval instant at 4m limitk(1, mixed{instance="1"})
+  {__name__="mixed", instance="1"} 10
+
+eval instant at 5m limitk(1, mixed{instance="1"})
+  {__name__="mixed", instance="1"} 10
+
+eval instant at 3m count(limitk(2, mixed))
+  {} 2
+
+eval instant at 4m count(limitk(2, mixed))
+  {} 2
+
+eval instant at 5m count(limitk(2, mixed))
+  {} 2
+
+eval range from 0m to 5m step 1m limitk(1, mixed{instance="1"})
+  {__name__="mixed", instance="1"} {{schema:0 sum:20 count:20}} {{schema:0 sum:20 count:20}} {{schema:0 sum:20 count:20}} {{schema:0 sum:20 count:20}} 10 10
+
+eval range from 0m to 5m step 1m limitk(1, mixed{instance="2"})
+  {__name__="mixed", instance="2"} 10 10 10 10 {{schema:0 sum:20 count:20}} {{schema:0 sum:20 count:20}}
+
+eval range from 0m to 5m step 1m count(limitk(2, mixed))
+  {} 2 2 2 2 2 2
+
+eval range from 0m to 5m step 1m count(limitk(2, mixed))
+  {} 2 2 2 2 2 2
+
+eval range from 0m to 5m step 1m count(limitk(scalar(foo), mixed))
+  {} 1 2 1 2 1 2
+
+clear
+
+load 1m
+  empty{instance="1"} _ _ _ _ _ _
+  empty{instance="2"} _ _ _ _ _ _
```

Comment:
If there are no samples, then the selector will return no series (rather than a series with no samples).

If you want to exercise the case where a series is present but has no samples, you'll need to do something to filter the samples after they've been loaded from storage - for example, populate the series with samples all of value `1` and then filter them out with something like `count(limitk(2, empty > 1))`.